src/checkers/inference/solver/util/PrintUtils.java

```diff
@@ -0,0 +1,184 @@
+        StringBuilder printResult = new StringBuilder();
+
+        System.out.println("/***********************Results****************************/");
+        for (Integer j : result.keySet()) {
+            printResult.append("SlotID: ");
+            printResult.append(String.valueOf(j));
+            for (int i = 0; i < maxLength + 2 - String.valueOf(j).length(); i++) {
+                printResult.append(" ");
+            }
+            printResult.append("Annotation: ");
+            printResult.append(result.get(j).toString());
+            printResult.append("\n");
+        }
+        System.out.println(printResult.toString());
+        System.out.flush();
+        System.out.println("/**********************************************************/");
+    }
+
+    public static void writeResult(Map<Integer, AnnotationMirror> result) {
+        
+        final int maxLength = String.valueOf(InferenceMain.getInstance().getSlotManager().getNumberOfSlots()).length();
+        StringBuilder printResult = new StringBuilder();
+
+        for (Integer j : result.keySet()) {
+            printResult.append("SlotID: ");
+            printResult.append(String.valueOf(j));
+            for (int i = 0; i < maxLength + 2 - String.valueOf(j).length(); i++) {
+                printResult.append(" ");
+            }
+            printResult.append("Annotation: ");
+            printResult.append(result.get(j).toString());
+            printResult.append("\n");
+        }
+
+        File basePath = new File(new File("").getAbsolutePath());
+        String writePath = basePath.getAbsolutePath() + "/result" + ".txt";
+        File file = new File(writePath);
+        PrintWriter pw;
+        try {
+            pw = new PrintWriter(file);
+            pw.write(printResult.toString());
+            pw.close();
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+
+        System.out.println("Result has been written to: " + writePath);
+    }
+
+    private static StringBuilder buildStatistic(Map<StatisticKey, Long> statistic,
+            Map<String, Integer> modelRecord, String backEndType, boolean useGraph,
+            boolean solveInParallel) {
+
+        StringBuilder statisticsText = new StringBuilder();
+        StringBuilder basicInfo = new StringBuilder();
+        StringBuilder timingInfo = new StringBuilder();
+
+        // Basic info
+        buildStatisticText(statistic, basicInfo, StatisticKey.SLOTS_SIZE);
+        buildStatisticText(statistic, basicInfo, StatisticKey.CONSTRAINT_SIZE);
+        for (Map.Entry<String, Integer> entry : modelRecord.entrySet()) {
+            buildStatisticText(entry.getKey(), entry.getValue(), basicInfo);
+        }
+        if (useGraph) {
+            buildStatisticText(statistic, basicInfo, StatisticKey.GRAPH_SIZE);
+        }
+
+        if (backEndType.equals(Constants.MAX_SAT) || backEndType.equals(Constants.LINGELING)) {
```

Comment:
Just a reminder: I think later on we need refactor the logic here to let it be more extendable for adding new backends. Specifically, currently this class is highly coupling with specific backEnd types (e.g. MAX_SAT, LINGELING), which means, if we add new backends that have customize statistic keys, we need add more if-else branch here, which would eventually make code here hard to read and maintain. Thus, it is nice to decoupling this relationship.